static/app/components/events/interfaces/spans/spanContext.tsx

```diff
@@ -0,0 +1,67 @@
+import {Component, createContext} from 'react';
+
+import {ProcessedSpanType} from './types';
+
+export type SpanContextProps = {
+  addExpandedSpan: (span: Readonly<ProcessedSpanType>, callback?: () => void) => void;
+  didAnchoredSpanMount: boolean;
+  isSpanExpanded: (span: Readonly<ProcessedSpanType>) => boolean;
+  markAnchoredSpanIsMounted: () => void;
+  removeExpandedSpan: (span: Readonly<ProcessedSpanType>, callback?: () => void) => void;
+};
+
+const SpanContext = createContext<SpanContextProps>({
+  didAnchoredSpanMount: false,
+  markAnchoredSpanIsMounted: () => undefined,
+  addExpandedSpan: () => undefined,
+  removeExpandedSpan: () => undefined,
+  isSpanExpanded: () => false,
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export class Provider extends Component<Props> {
+  didAnchoredSpanMount = false;
+
+  // This set keeps track of all spans which are currently expanded to show their details.
+  // Since the span tree is virtualized, we need this so the tree can remember which spans have been expanded after they unmount
+  expandedSpansMap: Set<Readonly<ProcessedSpanType>> = new Set();
+
+  markAnchoredSpanIsMounted = () => {
```

Comment:
We keep track of whether the anchored span has mounted for the first time. It's necessary to keep track of this, otherwise we'd keep scrolling to the anchored span every time it mounts again